### ISSUE TITLE
Color Palette doesn't seem to be working

### ISSUE THREAD

**Issue by rmiessle** [sgsinclair/Voyant#426]
I haven't been able to get the color palette option to work, it always seems to just remain on the default colors regardless if I select one of the other palettes, or try to make a custom one. Browser doesn't seem to matter, and same behavior in the 2.4 M7 VoyantServer release.

**Comment by ajmacdonald** [comment]
I think what's happening is that terms get assigned colours initially, but changing to a different palette doesn't overwrite those colours.
Here's an example where I switched to a custom red, green, blue palette and then increased the term size in Cirrus. As you can see, the new terms get assigned colours from the custom palette:
![palettes](https://user-images.githubusercontent.com/882444/42233596-0596cd0e-7ec0-11e8-8518-c2f53c91b33b.png)


@sgsinclair it probably makes sense to remove previous term/colour associations when changing palettes right?

**Comment by sgsinclair** [comment]
Yes, resetting is probably easiest. It might be possible to store a map of words to palette index position instead of words to colours (then changing palettes would simply use the same index but in a different palette), but that might cause other problems.

**Comment by rmiessle** [comment]
Ok, just making sure I wasn't missing something completely obvious.

**Comment by ajmacdonald** [comment]
@sgsinclair I've looked into it and the only problem I can see is in the CategoriesManager where the user can set the category feature (and therefore the term/colour association) to any arbitrary colour: https://github.com/sgsinclair/Voyant/blob/master/src/main/webapp/app/util/CategoriesManager.js#L128-L139

**Comment by agatamer** [comment]
Dear Sork,
Thank you so much! I will enrol right away then.
Thank you!
Regards,
Agata

---- Original Message ----

**Comment by ajmacdonald** [comment]
@sgsinclair so it looks like this issue was introduced when I commented out [this line](https://github.com/sgsinclair/Voyant/commit/7d61999244d30812b8afa84b347325a61bd4e084#diff-55786c01cfea29b33d84322340f3700fR254) in order to get categories working.

Closing the categories builder saves the category colours to the term/colour associations. But then closing the options dialog afterwards will either keep or wipe those associations (depending on the existence of the above line). In the case of the categories we want to keep the associations but in the case of just changing to a different palette we want to wipe them.

It seems like it may require re-working categories...
